modules/imgproc/src/resize.cpp

```diff
@@ -1104,6 +1104,79 @@ resizeNN( const Mat& src, Mat& dst, double fx, double fy )
     }
 }
 
+class resizeNNPILInvoker :
+    public ParallelLoopBody
+{
+public:
+    resizeNNPILInvoker(const Mat& _src, Mat &_dst, int *_x_ofs, double _ify) :
+        ParallelLoopBody(), src(_src), dst(_dst), x_ofs(_x_ofs),
+        ify(_ify)
+    {
+    }
+
+    virtual void operator() (const Range& range) const CV_OVERRIDE
+    {
+        Size ssize = src.size(), dsize = dst.size();
+        int y, x, pix_size = (int)src.elemSize();
+        double yo = ify / 2;
+
+        for( y = range.start; y < range.end; y++ )
+        {
+            int yi;
+            if (yo < 0)
+                yi = -1;
+            else
+                yi = (int)yo;
+            if(yi >= 0 && yi < ssize.height)
+            {
+                for( x = 0; x < dsize.width; x++ )
+                {
+                    dst.at<char>(y, x) = src.at<char>(yi, x_ofs[x]);
```

Comment:
You support only `CV_8UC1`. What about other types of `Mat`? You should at least add `Not implemented` assertion.

Reply:
I've found in resize.cpp similar situation:
if (interpolation == INTER_LINEAR_EXACT && (_src.depth() == CV_32F || _src.depth() == CV_64F))
        interpolation = INTER_LINEAR; // If depth isn't supported fallback to generic resize

and add my code:
if (interpolation == INTER_NEAREST_PIL && _src.depth() != CV_8UC1)
        interpolation = INTER_NEAREST; // INTER_NEAREST_PIL is supported only for grayscale images

If image is not a mask (binary) or grayscale  generic INTER_NEAREST will be used. I suppose it could be better than Not implemented assertion. What do you think?
In case NotImplemented is a demand - I'll do it.

Reply:
Case with `INTER_LINEAR_EXACT` is slightly different, it is same algorithm as `INTER_LINEAR` but with stable calculation method which produces same results on all platforms. While `INTER_NEAREST_PIL` is separate algorithm and can produce completely different results so the assertion is more appropriate.

Reply:
OK, I've added 
if (interpolation == INTER_NEAREST_PIL && _src.depth() != CV_8UC1)        
        CV_Error(Error::StsNotImplemented, "INTER_NEAREST_PIL supports only CV_8UC1");

Reply:
does not look very fast.
```
const char* srcrow = src.ptr<char>(yi);
char* dstrow = dst.ptr<char>(y);
for(x = 0; x < dsize.width; x++)
    dstrow[x] = srcrow[x_ofs[x]];
```
should be a bit better